### ISSUE TITLE
THis does not have all hidden APIs

### ISSUE THREAD

**Issue by championswimmer** [anggrayudi/android-hidden-api#19]
Basically android.jar is not **only** framework.jar, but also contains classes from a few other common libraries

We at [AOKP](http://aokp.co) need internal APIs to build some of our apps, so we have a script to generate android.jar with all methods.

As you can see in the script here 
https://github.com/AOKP/build/blob/mm/tools/customsdkgen.sh#L16
all these classes need to be present in the android.jar 
basically this is spread out across    
 - framework.jar
 - framework2.jar
 - core.jar
 - telephony.jar

**Comment by championswimmer** [comment]
Also compiling this from BlissPop is a bad idea. It will have some methods and fields that not all Android phones have (custom methods and fields Bliss devs have added)

**Comment by Trumeet** [comment]
Hi, just see #36 , it adds jars below.

**Comment by anggrayudi** [comment]
This is now addressed. You're right that `android.jar` must be more than just `framework.jar` — the new **`hiddenjar`** CLI merges `framework.jar` **plus every other `$BOOTCLASSPATH` jar** (core, telephony, and the modern mainline/APEX modules), which is exactly the completeness you described:

```bash
./cli/hiddenjar build --api <level>
# or via Gradle:
./gradlew buildHiddenJar -Papi=<level>
```

Unlike the old approach it merges the whole `$BOOTCLASSPATH` (including mainline/APEX modules such as Wi‑Fi, Bluetooth, connectivity and media), not just `framework.jar`, and it verifies the result actually compiles a hidden API before finishing. Details: the README section **“Build Your Own Custom `android.jar` (CLI)”** and `docs/BUILD_CUSTOM_ANDROID_JAR_FROM_EMULATOR.md`.

Closing — reopen if you find something still missing.
